File: leetcode/sort-colors.cpp
#include <bits/stdc++.h>

using namespace std;

class Solution
{
 public:
    void sortColors(vector<int>& nums)
    {
        // use count sort
        auto size = nums.size();
        int counter[] = {0, 0, 0};

        for (auto n : nums) {
            ++counter[n];
        }

        for (auto i = 0, j = 0; i < size; ++i) {
            while (counter[j] <= 0) {
                ++j;
            }
            nums[i] = j;
            --counter[j];
        }
    }
};

File: leetcode/sort-colors.py
class Solution:
    def sortColors(self, nums: List[int]) -> None:
        """
        Do not return anything, modify nums in-place instead.
        """
        length = len(nums)

        last_index = 0
        for i in range(length):
            if nums[i] == 0:
                nums[i], nums[last_index] = nums[last_index], nums[i]
                last_index += 1

        for i in range(length):
            if nums[i] == 1:
                nums[i], nums[last_index] = nums[last_index], nums[i]
                last_index += 1
